moshi/moshi/models/tts.py

```diff
@@ -0,0 +1,624 @@
+
+
+DEFAULT_DSM_TTS_REPO = "alex-testing/dsm-tts-rc"
+DEFAULT_DSM_TTS_VOICE_REPO = 'alex-testing/dsm-tts-rc-voices'
+
+
+@dataclass
+class TokenIds:
+    """
+    The token ids for special tokens:
+        - word: a new word is starting.
+        - pad: padding, nothing happens.
+        - main: indicates the start of turn of the main speaker.
+        - other: indicates the start of turn of the other speaker.
+        - zero: special value that is embedded to exactly 0.
+        - ungenerated: indicate that a value is not yet generated but should be
+        - card: text cardinality, including the initial token (1 + tokenizer cardinality).
+            This is used for multiplexing multiple input tokens into the text stream.
+
+    """
+    word = 0
+    pad = 3
+    main = 1
+    other = 2
+    zero = -1
+    ungenerated = -2
+    card: int = 8001
+
+
+@dataclass
+class Entry:
+    """One word to generate.
+
+    Args:
+        tokens: list of tokens for this word.
+        word: word as string.
+        padding: if > 0, we will force that many padding after the word.
+            To have an effect this should be more than len(tokens), as we will
+            prevent new word until we have passed all the text tokens.
+        audio_tokens: is used when some audio should be used as a prefix in the model."""
+    tokens: list[int]
+    word: str
+    padding: int = 0
+    audio_tokens: torch.Tensor | None = None
+
+
+@dataclass
+class State:
+    """State of the TTS Machine.
+
+    Args:
+        entries: queue containing the entries to generate.
+        remaining_padding: how many times the model can still sample a pad.
+        forced_padding: how many times the model is still forced to sample a pad.
+        queued: queue containing the main stream text tokens to feed.
+        lookahead: queue containing the lookahead text tokens to feed.
+        end_step: once we reach the end of the generation, this is set to the current step.
+            The end of the generation is once the model samples a `word` but `entries` is empty.
+        consumption_times: list of steps at which each entry in `entries` was consumed.
+        transcript: list of tuples `(word, step)`, at which each word was consumed.
+        audio_tokens_remaining: when using some audio as prefix, this would contain the remaining
+            audio tokens to force into the model.
+        zero_text_remaining: when using an audio only prefix, for how long we should still force
+            the text to be `zero`.
+    """
+    entries: deque[Entry]
+    remaining_padding: int
+    forced_padding: int
+    queued: deque[int] = field(default_factory=deque)
+    lookahead: deque[int] = field(default_factory=deque)
+    end_step: int | None = None
+    consumption_times: list[int] = field(default_factory=list)
+    transcript: list[tuple[str, int]] = field(default_factory=list)
+
+    def get_tokens_ahead(self, lookahead: int) -> list[int]:
```

Comment:
Having both a parameter named `lookahead` and a `self.lookahead` field is a bit confusing to me, maybe you could rename these e.g. `lookahead_tokens`.